parser.js

```diff
@@ -941,11 +941,20 @@
 
     function parseIdentityEscape() {
       // IdentityEscape ::
-      //      SourceCharacter but not c
+      //      [+U] SyntaxCharacter
+      //      [+U] /
+      //      [~U] SourceCharacterIdentityEscape[?N]
+      // SourceCharacterIdentityEscape[?N] ::
+      //      [~N] SourceCharacter but not c
+      //      [+N] SourceCharacter but not one of c or k
```

Comment:
The gramma states `not one of c or k`. From reading the code it looks only the case `not c` is handled. Is this correct? If yes, can you please add the handling for `or k` as well?

Please add a unit test for the `or k` part.

Reply:
Thank you! Addressed in 2cf92b1.

I thought `parseNamedReference` would process `\\k` and then throw but apparently I am wrong.